server/src/routers/api/userProfileRouter.js

```diff
@@ -0,0 +1,39 @@
+const express = require("express");
+const router = express.Router();
+const UserProfile = require("../../models/Profile");
+
+// @route GET api/profile/me
+// @desc Get your own user profile
+// @access Private
+
+router.get("/me", async (req, res) => {
+  try {
+    const userProfile = await UserProfile.findOne({
```

Comment:
This is a private route so include auth middleware,
router.get("/me", auth, async(req,res)=>{
// here user who is trying to access this route is available at req.user now ( this was done in the auth middleware have a look at it)
});